### ISSUE TITLE
Tied note lengths not accepted as note lengths in tempo declarations

### ISSUE THREAD

**Issue by elydpg** [alda-lang/alda-core#57]
When declaring a tempo, it was noted that it's possible to use dotted notes as note lengths like so:

```
(tempo "4." 120) c d e f g
```

However, using a tied note throws the following:
```
(tempo "4~16" 120) c d e f g
[27713] ERROR java.lang.Exception: Invalid note length: 4~16, compiling:(null:1:1)
```

**Comment by daveyarwood** [comment]
Nice catch. This is definitely a bug.

Probably what we should do is split the string on `~`, parse each component as a note length (failing if it isn't valid), and then add them together.

cc: @pzxwang

**Comment by elydpg** [comment]
Hm. Anything else count as a syntactic note length? I think that covers everything

**Comment by daveyarwood** [comment]
I think that's it, as far as standard note lengths go in Alda.

Alda also has second and millisecond durations, but I don't think it really makes sense to support those here. I can't imagine someone would ever want to specify a duration of 180 bpm, where one beat is exactly 532 ms long, for instance 😄

**Comment by elydpg** [comment]
>I think that's it, as far as standard note lengths go in Alda.

>Alda also has second and millisecond durations, but I don't think it really makes sense to support those here. I can't imagine someone would ever want to specify a duration of 180 bpm, where one beat is exactly 532 ms long, for instance 😄

Actually that wouldn't work, because if beats were specified in milliseconds it would predetermine the number of beats per minute. Maybe a single String taking a millisecond or second value would suffice

**Comment by daveyarwood** [comment]
Now that I think about it, tempo is, by definition, agnostic to the physical duration of a note.

So, we definitely can't support the kind of thing I just described. It's nonsensical.

**Comment by elydpg** [comment]
Yep. The single argument millisecond tempo function would work, but that's your call.

**Comment by daveyarwood** [comment]
Ah, I see what you're saying. Something like `(tempo! "532ms")` or something like that, where you're saying that one beat lasts 532 milliseconds.

That's an interesting thought, but not something I think we need.

**Comment by daveyarwood** [comment]
Closed via #58. Will deploy a new release of Alda soon with the fix.
